File: Python/binary_prefix_divisible_by_5.py
def prefixesDivBy5(A: list) -> list:
    M = 0
    result = []
    for num in A:
        M = M * 2 + num
        if M % 5 == 0:
            result.append(True)
        else:
            result.append(False)
    return result


if __name__ == "__main__":
    print(prefixesDivBy5([1, 1, 1, 0, 1]))

File: Java/src/com/dbc/algorithm/Daily/BinaryPrefixDivisibleBy5.java
package com.dbc.algorithm.Daily;

import java.util.ArrayList;
import java.util.List;

public class BinaryPrefixDivisibleBy5 {
    public static List<Boolean> prefixesDivBy5(int[] A) {
        int M = 0;
        List<Boolean> result = new ArrayList<>();
        for (int num : A) {
            M = ((M << 1) + num) % 5;
            result.add(M == 0);
        }

        return result;
    }
}
